05-part_two.Rmd

```diff
@@ -75,7 +73,7 @@ To implement unit tests, it is generally easiest to use a unit testing framework
 
 The `testthat` package, maintained by the RStudio team, is commonly used for unit testing in R. Although most of its documentation and features focus on testing in the context of package writing, `testthat` can also be used with regular R projects.  There's a brief overview of unit testing below, but for a more detailed introduction see [here](https://katherinemwood.github.io/post/testthat/) or [here](https://www.r-bloggers.com/automated-testing-with-testthat-in-practice/) for a more in-depth overview of automated testing, including test driven development. 
 
-In `testthat`, tests are called **expectations**. These spell out the expected behaviour of your function or unit of code. `testthat` provides many functions to define expectations - see details [here](http://r-pkgs.had.co.nz/tests.html). It is good practice to save tests relating to the different parts of your code in different files, all saved within a dedicated testing directory. Below is an example of testing a simple function.
+In `testthat`, tests are called **expectations**. These spell out the expected behaviour of the function or unit of code. `testthat` provides many functions to define expectations - see details [here](http://r-pkgs.had.co.nz/tests.html). It is good practice to save tests relating to the [different code files](#code_structure) in different test files, all saved within a dedicated testing directory. Below is an example of testing a simple function.
 
 Firstly we need to have a defined function to test:
```

Comment:
There's a typo lower down that I can't tag where you've said:  Then we can defined ... so the d needs dropping so it's just define